Add Chisel for UB, erebus, atum, lightningcraft, GC ore variants

KleinGrrmpf opened the pull request:
Also fixes two typos in the comments lol
I also remembered that you can change through the ore types in the Ore Dictionarificator, so maybe this is not needed then... Welp... ¯\_(ツ)_/¯

WaitingIdly reviewed (changes requested):
another possible way to do this would be something like:
```zenscript
val chiselSet as IOreDictEntry[string] = [
    coalore: <ore:oreCoal>,
    ironore: <ore:oreIron>,
    // etc 
];

for name, ores in chiselSet {
    Carving.addGroup(name);
    for item in ores.items {
        Carving.addVariation(name, item);
    }
}

```
but i believe this might have the issue of allowing unobtainable and disabled ores to be obtained - would have to check each category to ensure this doesnt happen

KleinGrrmpf commented:
I knew there were easier ways. I'll try my best if you want this implemented and don't say that Ore Dictionaryficator is enough of a conversion method.

KleinGrrmpf commented on overrides/scripts/Chisel.zs:
```diff
@@ -243,6 +243,366 @@ for i in 0 to 16 {
+Carving.addGroup("UBChargedCertusQuartzOre");
+Carving.addVariation("UBChargedCertusQuartzOre", <appliedenergistics2:charged_quartz_ore>);
+for i in 0 to 8 {
+Carving.addVariation("UBChargedCertusQuartzOre", <undergroundbiomes:igneous_stone_appliedenergistics2_charged_quartz_ore>.withDamage(i)); 
```
Fixed

KleinGrrmpf commented on overrides/scripts/Chisel.zs:
```diff
@@ -243,6 +243,366 @@ for i in 0 to 16 {
+Carving.addVariation("UBGoldOre", <abyssalcraft:abygolore>);
+Carving.addVariation("UBGoldOre", <atum:gold_ore>);
+Carving.addVariation("UBGoldOre", <erebus:ore_gold>);
+Carving.addVariation("UBGoldOre", <lightningcraft:ore_block>.withDamage(1));
```
Fixed

KleinGrrmpf commented:
Not sure if this is a problem but I added conversion between Oil Sand and Oil Shale, which technically give different secondary outputs. Sand gives Tar and Shale gives Flint. So if you want this taken out, tell me. I just added them together because the UB variants look exactly the same. However, I'd love if it was possible to use the UB gravel textures for the Oil Shale variants. Not sure if that is possible tho.

KleinGrrmpf commented:
Ok, so your approach worked partly, Waiting. It added all the ores that aren't UB ores. But those ores are either created after the recipes are created or there is some other reason that they aren't implemented. But the weird thing is that it works if I add all the chisel recipes manually...

This is the part of code that I used:

`//Underground Biomes Ore Conversion
val chiselSet as IOreDictEntry[string] = {
    CoalOre : <ore:oreCoal>,
    IronOre : <ore:oreIron>,
    GoldOre : <ore:oreGold>,
    DiamondOre : <ore:oreDiamond>,
    EmeraldOre : <ore:oreEmerald>,
    LapisOre : <ore:oreLapis>,
    RedstoneOre : <ore:oreRedstone>,
    QuartzOre : <ore:oreQuartz>,
    ChargedCertusQuartzOre : <ore:oreChargedCertusQuartz>,
    BlackQuartzOre : <ore:oreQuartzBlack>,
    DarkOre : <ore:oreDark>,
    AquamarineShale : <ore:oreAquamarine>,
    CinnabarOre : <ore:oreCinnabar>,
    AmberBearingStone : <ore:oreAmber>,
    GarnetOre : <ore:oreGarnet>,
    OpalOre : <ore:oreOpal>,
    AmethystOre : <ore:oreAmethyst>,
    CoraliumOre : <ore:oreCoralium>,
    InferiumOre : <ore:oreInferium>,
    ProsperityOre : <ore:oreProsperity>,
    RealmiteOre : <ore:oreRealmite>,
    RupeeOre : <ore:oreRupee>,
    ArlemiteOre : <ore:oreArlemite>,
    YelloriteOre : <ore:oreYellorite>,
    OsmiumOre : <ore:oreOsmium>,
    DraconiumOre : <ore:oreDraconium>,
    DestabilizedRedstoneOre : <ore:oreClathrateRedstone>,
    CopperOre : <ore:oreCopper>,
    TinOre : <ore:oreTin>,
    SilverOre : <ore:oreSilver>,
    LeadOre : <ore:oreLead>,
    AluminumOre : <ore:oreAluminum>,
    NickelOre : <ore:oreNickel>,
    IridiumOre : <ore:oreIridium>
};

for name, ores in chiselSet {
    Carving.addGroup(name);
    for item in ores.items {
        Carving.addVariation(name, item);
    }
}

    //CertusQuartzOre: <ore:oreCertusQuartz>,
    //OilSand: <ore:oreClathrateOilSand>,
    //DimensionalShardOre : <ore:oreDimensionalShard>,`

The three not included have more ores in their OreDictionary. And I haven't found a solution for the Dimensional Shard Ore yet...

KleinGrrmpf commented:
Now I don't know what to do anymore. I tried adding the oreDictionaries to the ores in the UB config file, hoping that this would fix the issue of crafttweaker not recognizing the oreDictionaries on the UB ores but that didn't work either. If you have any ideas what I could still try, I'd be open for anything. Or I would have to add the chisel groups all manually like I did in the first commit.

KleinGrrmpf commented:
> Not sure if this is a problem but I added conversion between Oil Sand and Oil Shale, which technically give different secondary outputs. Sand gives Tar and Shale gives Flint. So if you want this taken out, tell me. I just added them together because the UB variants look exactly the same. However, I'd love if it was possible to use the UB gravel textures for the Oil Shale variants. Not sure if that is possible tho.

And having the background texture of an ore being gravel isn't possible from what I found out

KleinGrrmpf commented:
Ok, so I tested it and it all seemed to work. When I loaded up my testworld, it told me that there were missing Iridium Ore Blocks and Items, so removing them from UB worked, there was no error, so removing the recipes from the Mekanism Machines worked too, there were no additional AE2 facades, so removing those worked as well and chiseling worked for all the ores too.

I'd still suggest removing one set of the Oil Sands from the Chisel Group. Because right now when you chisel normal Oil Sand or the Red Oil Sand, you can chisel them into each other as well as all the UB variants of both of the Sands. And since they have the exact same texture, this doesn't provide anything useful, it only clutters the Chisel UI. So I'd suggest only adding the UB ores from the normal Oil Sand plus the normal and Red Oil Sand to the chisel group. And since with the next update Oil Sand and Red Oil Sand will hopefully always drop the TF variants, no matter what UB variant you mined, there won't be any unchiselable Oil Sands around.
